# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerDebuggerManager.h
# Commit: e80b18ca9165
# Full Hash: e80b18ca916518c73d5435fe8f842d9f0e1b7585
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-25 21:02:33
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r cb743703a9ee -r e80b18ca9165 dom/workers/WorkerDebuggerManager.h
--- a/dom/workers/WorkerDebuggerManager.h	Tue Mar 25 09:11:22 2025 +0000
+++ b/dom/workers/WorkerDebuggerManager.h	Tue Mar 25 09:13:20 2025 +0000
@@ -14,15 +14,13 @@
 #include "nsDebug.h"
 #include "nsIObserver.h"
 #include "nsISupports.h"
+#include "nsIWorkerDebugger.h"
 #include "nsIWorkerDebuggerManager.h"
 #include "nsTArray.h"
+#include "nsXULAppAPI.h"
 
-#define WORKERDEBUGGERMANAGER_CID                    \
-  {                                                  \
-    0x62ec8731, 0x55ad, 0x4246, {                    \
-      0xb2, 0xea, 0xf2, 0x6c, 0x1f, 0xe1, 0x9d, 0x2d \
-    }                                                \
-  }
+#define WORKERDEBUGGERMANAGER_CID \
+  {0x62ec8731, 0x55ad, 0x4246, {0xb2, 0xea, 0xf2, 0x6c, 0x1f, 0xe1, 0x9d, 0x2d}}
 #define WORKERDEBUGGERMANAGER_CONTRACTID \
   "@mozilla.org/dom/workers/workerdebuggermanager;1"
 
@@ -39,7 +37,7 @@
   nsTArray<nsCOMPtr<nsIWorkerDebuggerManagerListener>> mListeners;
 
   // Only touched on the main thread.
-  nsTArray<RefPtr<WorkerDebugger>> mDebuggers;
+  nsTArray<nsCOMPtr<nsIWorkerDebugger>> mDebuggers;
 
  public:
   static already_AddRefed<WorkerDebuggerManager> GetInstance();
@@ -63,13 +61,19 @@
   void RegisterDebuggerMainThread(WorkerPrivate* aWorkerPrivate,
                                   bool aNotifyListeners);
 
+  void RegisterDebugger(nsIWorkerDebugger* aRemoteWorkerDebugger);
+
   void UnregisterDebugger(WorkerPrivate* aWorkerPrivate);
 
   void UnregisterDebuggerMainThread(WorkerPrivate* aWorkerPrivate);
 
+  void UnregisterDebugger(nsIWorkerDebugger* aRemoteWorkerDebugger);
+
   uint32_t GetDebuggersLength() const;
 
-  WorkerDebugger* GetDebuggerAt(uint32_t aIndex) const;
+  nsIWorkerDebugger* GetDebuggerAt(uint32_t aIndex) const;
+
+  nsCOMPtr<nsIWorkerDebugger> GetDebuggerById(const nsString& aWorkerId);
 
  private:
   nsTArray<nsCOMPtr<nsIWorkerDebuggerManagerListener>> CloneListeners();